src/pages/forms/createAccount.js

```diff
@@ -0,0 +1,321 @@
+import React, { useEffect, useRef, useState } from "react";
+import styles from "./createAccount.module.css";
+import { Link } from "react-router-dom";
+// import UsePasswordToggle from "../customHooks/usePasswordToggle";
+// import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useParams } from "react-router-dom";
+const CreateAccount = () => {
+  const inputData1 = useRef();
+  const inputData2 = useRef();
+  // const inputData3 = useRef();
+  // const inputData4 = useRef();
+  const inputData5 = useRef();
+  const inputData6 = useRef();
+  const inputData7 = useRef();
+  // const [PasswordInputType, ToggleIcon] = UsePasswordToggle();
+
+  // console.log(PasswordInputType, ToggleIcon);
+
+  const [showEye, setShowEye] = useState(true);
+  const [showPass, setShowPass] = useState(true);
+  const [radioMarked, setRadioMarked] = useState(false);
+  const [randomVal1, setRandomVal1] = useState(true);
+  const [randomVal2, setRandomVal2] = useState(true);
+  const [emailVal, setEmailVal] = useState(false);
+  const [passVal, setPassVal] = useState();
+  // const [ag, setAg] = useState(false);
+  // const [jj, setJj] = useState(false);
+
+  // const [ol, setOl] = useState(false);
+  // const [bz, setBz] = useState(false);
+  const [firstNameVal, setFirstNameVal] = useState(false);
+  const [lastNameVal, setLastNameVal] = useState(false);
+  const [bussNameVal, setBussNameVal] = useState(false);
+  const params = useParams();
+
+  useEffect(() => {
+    if (params.id === "brand") {
+      setRadioMarked(true);
+      // setOl(true);
+    } else {
+      setRadioMarked(false);
+      // setBz(true);
+    }
+  }, []);
+
+  const onHil = () => {
+    setRadioMarked(true);
+    // setOl(true);
+    // setBz(false);
+    setRandomVal1(true);
+    setRandomVal2(false);
+  };
+
+  const onBil = () => {
+    setRadioMarked(false);
+    // setBz(true);
+    // setOl(false);
+
+    setRandomVal1(false);
+    setRandomVal2(true);
+  };
+
+  const onEy = () => {
+    setShowEye(false);
+    setShowPass(false);
+  };
+
+  const onFy = () => {
+    setShowEye(true);
+    setShowPass(true);
+  };
+
+  const sumtHandler = (e) => {
+    e.preventDefault();
+
+    if (
+      inputData5.current.value.includes("_") ||
+      inputData5.current.value.includes("-") ||
+      inputData5.current.value.length < 1
+    ) {
+      setFirstNameVal(true);
+    } else {
+      setFirstNameVal(false);
+    }
+
+    if (
```

Comment:
in all the places , handle validation properly . This is not the correct way. Check for form validations in react form

Reply:
this code is not optimised . Checkk how to handle form validations .